Sources/TensorFlow/Core/LazyTensorTrace.swift

```diff
@@ -89,13 +89,55 @@ class LazyTensorTraceBuilder {
         return materializationTraceInfo([lazyOperation])
     }
 
+    /// Trace the given function and return the trace.
+    static func trace<In: TensorGroup, Out: TensorGroup>(_ fn: (In) -> Out) -> LazyTensorTrace {
+        precondition(_RuntimeConfig.useLazyTensor, "Lazy tensor is not enabled for tracing.")
+
+        // Set up inputs for running `fn`
+        let inputOps = In._typeList.map { Self.makePlaceholder(with: $0) }
+        let inputHandles = inputOps.map { LazyTensorHandle(_lazy: $0, index: 0) }
+        let input = In(_handles: inputHandles)
+
+        // Run the function.
+        let output: TensorArrayProtocol = fn(input)
+
+        // Set up the closure that determines if a `LazyTensorOperation` should be an output.
+        let outputLazyOperations = output._tensorHandles.map {
+            (handle: _AnyTensorHandle) -> LazyTensorOperation in
+            let lazyOp = lazyTensorOperation(handle)
+            precondition(lazyOp != nil, "Found a non-lazy tensor in output when tracing.")
+            return lazyOp!
+        }
+        let outputIds = Set<ObjectIdentifier>(outputLazyOperations.map { ObjectIdentifier($0) })
+        let isOutput: (LazyTensorOperation) -> Bool = { outputIds.contains(ObjectIdentifier($0)) }
+
+        // Create the builder and get the trace.
+        let builder = LazyTensorTraceBuilder()
+        builder.neverPromoteConstants = true
+        builder.isOutput = isOutput
+        /// Set up the inputs for the builder as we need to have them in a specific order.
```

Comment:
Use 2 slashes for non-documentation comments.
```suggestion
        // Set up the inputs for the builder as we need to have them in a specific order.
```